### PR TITLE
k8s runner

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

k8s runner

Requires: https://github.com/ansible-collections/kubernetes.core/pull/307

##### ISSUE TYPE

- New Module Pull Request

### CODE REVIEW

**Comment by alinabuzachis** [comment]
recheck

File: plugins/module_utils/k8s/runner.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        module.fail_json(
+            msg="kubernetes-validate python library is required to validate resources"
+        )
+
+    if errors and module.params["validate"]["fail_on_error"]:
+        module.fail_json(
+            msg="\n".join([_prepend_resource_info(resource, error) for error in errors])
+        )
+    else:
+        return [_prepend_resource_info(resource, msg) for msg in warnings + errors]
```
Comment (Akasurde) [inline plugins/module_utils/k8s/runner.py:50]:
```suggestion
    if errors and module.params["validate"]["fail_on_error"]:
        module.fail_json(
            msg="\n".join([_prepend_resource_info(resource, error) for error in errors])
        )
    return [_prepend_resource_info(resource, msg) for msg in warnings + errors]
```

Comment (alinabuzachis) [reply plugins/module_utils/k8s/runner.py]: @Akasurde Thank you. Done! This requires https://github.com/ansible-collections/kubernetes.core/pull/307 to be merged first.

**Review by gravesm — APPROVED** [review]
LGTM. Once the Service PR merges and this passes we can go ahead and gate this.

**Pull request merged**